scripts/install.js

```diff
@@ -30,47 +32,60 @@ if (!skipFetch) {
 
         log.info('NAPA_INSTALL', 'completed successfully by download.');
         skipBuild = true;
-    } catch (e) {
-		errorCode = e.status;
-		
-		log.warn('NAPA_INSTALL', 'failed to download the pre-build binaries.');
-		if (!skipBuild) {
-			log.warn('NAPA_INSTALL', 'will fallback to build stage.');
-		}
+    }
+    catch (e) {
+        errorCode = e.status;
+
+        log.warn('NAPA_INSTALL', 'failed to download the pre-build binaries.');
+        if (!skipBuild) {
+            log.warn('NAPA_INSTALL', 'will fallback to build stage.');
+        }
     }
 }
 
 // ==== Try to build from sources ====
 if (!skipBuild) {
+    errorCode = 0;
+
     try {
         log.info('NAPA_INSTALL', 'building from sources...');
         execSync(buildCommand, { 'stdio': 'inherit' });
 
         log.info('NAPA_INSTALL', 'completed successfully by build.');
-    } catch (e) {
-		errorCode = e.status;
-		
-		log.warning('NAPA_INSTALL', 'failed to build from sources.');
+    }
+    catch (e) {
+        errorCode = e.status;
+
+        log.warning('NAPA_INSTALL', 'failed to build from sources.');
     }
 }
 
 // ==== Compile Typescript files ====
```

Comment:
enahnce to comment to specifically explain this is for supporting old npm versions.